torch/_higher_order_ops/flex_attention.py

```diff
@@ -1146,17 +1146,47 @@ def flex_attention_backward_fake_tensor_mode(
     torch.Tensor, torch.Tensor, torch.Tensor, tuple[Optional[torch.Tensor], ...]
 ]:
     with mode:
-        grad_query = torch.empty_like(query)
-        grad_key = torch.empty_like(key)
-        grad_value = torch.empty_like(value)
+        Bq = query.size(0)
+        Bkv = key.size(0)
+        Hkv = key.size(1)
+        seq_len_kv = key.size(2)
+        qk_head_dim = key.size(3)
+
+        grad_query = torch.empty_strided(
```

Comment:
What's the difference between this and empty_like? In particular, this can output a non-contiguous buffer?

Reply:
Exactly honestly, empty like semantics are weird and the memory format is a confusing args but tldr the strides not not match exactly

Reply:
🤔 if you do `torch.empty_like(query, memory_format=torch.preserve_format)` you get different strides?

It seems like with preserve_format (the default) we're supposed to desugar into empty_strided anyway: https://github.com/pytorch/pytorch/blob/main/aten/src/ATen/native/TensorFactories.cpp#L463

Reply:
Thanks Brian! I will use that for Grad Q